### ISSUE TITLE
Windows 10 LibreOffice ODS file not syncing: "Destination file exists"

### ISSUE THREAD

**Issue by craigarno** [owncloud/client#5324]
### Expected behaviour
- Windows LibreOffice 5.1.5.2 (x64) Calc File should be synced to Linux OwnCloud Server external RAID-5 storage space.
- If file can't be synced file should be ignored until file can be synced.
- Filename is Budget2016.ods and is "exclusive access" type spreadsheet, as opposed to a "shared access" spreadsheet.

### Actual behaviour
_Tell us what happens instead_
OwnCloud starts issuing Sync errors with the first "save" of the spreadsheet file.  LibreOffice remains open after all "Save" actions.  Save is accomplished using "CTRL-S" after updating spreadsheet.  This makes the file on disk "changed", but "still open" in LibreOffice.

### Steps to reproduce
OwnCloud Sync is connected using an HTTPS URL connection to the server.

1. Local Windows directory c:\Users\craig\Documents\Finances:
11/16/2016  12:59 AM            43,427 Budget2016.ods
09/19/2016  11:32 AM            42,764 Budget2016_conflict-20160826-100217 - Copy.ods
10/25/2016  11:09 AM            42,986 Budget2016_conflict-20161024-200107.ods
11/18/2016  09:04 AM            43,756 Budget2016_conflict-20161116-005912.ods

2. Remote Server directory /home/craig/Documents/Finances
-rwxr--r--  1 craig users    43427 Nov 16 00:59 Budget2016.ods

3. After "copy Budget2016_conflict-20161116-005912.ods Budget2016.ods"
OwnCloud.exe Windows produces the error: "Destination file exists"

Nothing critical gets deleted, this is good.  However, considerable effort must be expended with the mess OwnCloud Sync makes to locate which local file needs to be on the server, deleting and renaming local Windows files to make the Sync occur correctly.
My sense of it is that LibreOffice having the file open somehow screws up OwnCloud Sync.
If I shutdown OwnCloud Sync prior to Editing the local LibreOffice Calc file, start LibreOffice Calc and edit Budget2016.ods, then save/close LibreOffice, wait 5 minutes for Windows to let go of the file, then restart OwnCloud Sync... Sync occurs as I would expect and hope.

This behavior started with the last two releases of OwnCloud Sync.  3 versions back, OwnCloud Sync appeared to work fine with LibreOffice Calc files.  It doesn't now.

### Server configuration
**Operating system**:
Ubuntu 16.04 LTR

**Web server**: 
Apache/2.4.18 (Ubuntu) 

**Database**:
mysql  Ver 14.14 Distrib 5.7.13, for Linux (x86_64) using  EditLine wrapper
mysqlnd 5.0.12-dev - 20150407 - $Id: 241ae00989d1995ffcbbf63d579943635faf9972 $ 

**PHP version**:
PHP Version 7.0.8-0ubuntu0.16.04.2

**ownCloud version**:
ownCloud 8.2.8 (stable) 

**Storage backend (external storage)**:
Local RAID-5 Array:
> Filesystem       1K-blocks       Used             Available        Use%
> /dev/md1       23348263728 7490132516 14686030556  34%

### Client configuration
**Client version**:
Version 2.2.4 (build 6408).

**Operating system**:
Windows 10 Pro  v1607 OS Build 14393.447

**OS language**:
English US

**Installation path of client**:
C:\Program Files (x86)\ownCloud

### Logs

Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

```Template for output < 10 lines```

1. Client logfile: Output of `owncloud --logwindow` or `owncloud --logfile log.txt`
(On Windows using `cmd.exe`, you might need to first `cd` into the ownCloud directory)
(See also http://doc.owncloud.org/desktop/2.2/troubleshooting.html#client-logfile )

_--logwindow_
**11/18/2016 11:43:36 AM     Budget2016.ods    Documents\Finances    Server version downloaded, copied changed local file into conflict file.**

_--logfile_
Doesn't appear to be operating correctly.
- First run: produces a zero length file with the provided name
- Second run: produces this file:

>    11-18 12:29:34:612 0x4de7a10 OCC::Application::setupLogging: "################## ownCloud en_US (en_US) 2.2.4 (build 6408)"
> 11-18 12:29:34:613 0x4de7a10 OCC::Application::slotParseMessage: Running for 2762.87 sec
> 11-18 12:29:34:613 0x4de7a10 OCC::ownCloudGui::slotShowSettings: void OCC::ownCloudGui::slotShowSettings()
> 11-18 12:30:01:802 0x4de7a10 OCC::Folder::slotRunEtagJob: * Trying to check "https://oc.arno.com/remote.php/webdav/craig-Finances" for changes via ETag check. (time since last sync: 2784 s)
> 11-18 12:30:01:802 0x4de7a10 OCC::FolderMan::slotRunOneEtagJob: Scheduling "https://oc.arno.com/remote.php/webdav/craig-Finances" to check remote ETag
> 11-18 12:30:01:812 0x4de7a10 OCC::AbstractNetworkJob::start: !!! OCC::RequestEtagJob created for "https://oc.arno.com" + "/craig-Finances" "OCC::Folder"
> 11-18 12:30:31:799 0x4de7a10 OCC::Folder::slotRunEtagJob: * Trying to check "https://oc.arno.com/remote.php/webdav/craig-Finances" for changes via ETag check. (time since last sync: 2814 s)
> 11-18 12:30:31:799 0x4de7a10 OCC::FolderMan::slotRunOneEtagJob: Scheduling "https://oc.arno.com/remote.php/webdav/craig-Finances" to check remote ETag
> 11-18 12:30:31:800 0x4de7a10 OCC::AbstractNetworkJob::start: !!! OCC::RequestEtagJob created for "https://oc.arno.com" + "/craig-Finances" "OCC::Folder"

- Third run produces:

> 11-18 12:31:52:412 0x4de7a10 OCC::Application::setupLogging: "################## ownCloud en_US (en_US) 2.2.4 (build 6408)"
> 11-18 12:31:52:413 0x4de7a10 OCC::Application::slotParseMessage: Running for 2900.67 sec
> 11-18 12:31:52:413 0x4de7a10 OCC::ownCloudGui::slotShowSettings: void OCC::ownCloudGui::slotShowSettings()

2. Web server error log:

> 192.168.90.88 - craig [18/Nov/2016:13:03:32 -0800] "GET /ocs/v1.php/cloud/activity?page=0&pagesize=100&format=json HTTP/1.1" 200 3535
> 192.168.90.88 - craig [18/Nov/2016:13:04:02 -0800] "PROPFIND /remote.php/webdav/craig-Finances HTTP/1.1" 207 1237
> 192.168.90.88 - craig [18/Nov/2016:13:04:32 -0800] "PROPFIND /remote.php/webdav/craig-Finances HTTP192.168.90.88 - craig [18/Nov/2016:11:43:37 -0800] "GET /remote.php/webdav/craig-Finances/Budget201
> 6.ods HTTP/1.1" 200 44410
> 192.168.90.88 - craig [18/Nov/2016:11:43:37 -0800] "GET /remote.php/webdav/craig-Finances/Loan%20Pa
> yment%20Schedule.ods HTTP/1.1" 200 244232
> 192.168.90.88 - - [18/Nov/2016:11:43:59 -0800] "GET / HTTP/1.1" 200 4361
> 192.168.90.88 - - [18/Nov/2016:11:43:59 -0800] "GET /favicon.ico HTTP/1.1" 404 714
> 192.168.90.88 - - [18/Nov/2016:11:43:59 -0800] "GET /index.php HTTP/1.1" 200 2746
> 192.168.90.88 - - [18/Nov/2016:11:44:00 -0800] "GET /core/css/header.css?v=ef3ff5f0048ece99581a5e3c
> 39ddd79e HTTP/1.1" 200 2377
> 192.168.90.88 - - [18/Nov/2016:11:44:00 -0800] "GET /core/css/icons.css?v=ef3ff5f0048ece99581a5e3c3
> 9ddd79e HTTP/1.1" 200 1429

...

> 192.168.90.88 - craig [18/Nov/2016:12:59:46 -0800] "GET /ocs/v1.php/cloud/activity?page=0&pagesize=100&format=json HTTP/1.1" 200 3678
> 192.168.90.88 - craig [18/Nov/2016:12:59:48 -0800] "PROPFIND /remote.php/webdav/craig-Finances HTTP/1.1" 207 71790
> 192.168.90.88 - craig [18/Nov/2016:12:59:48 -0800] "PROPFIND /remote.php/webdav/craig-Finances/Citi HTTP/1.1" 207 2511
> 192.168.90.88 - craig [18/Nov/2016:12:59:49 -0800] "PROPFIND /remote.php/webdav/craig-Finances/receipts HTTP/1.1" 207 51661
> 192.168.90.88 - craig [18/Nov/2016:12:59:49 -0800] "GET /remote.php/webdav/craig-Finances/Budget2016.ods HTTP/1.1" 200 44410
> 192.168.90.88 - craig [18/Nov/2016:12:59:49 -0800] "GET /remote.php/webdav/craig-Finances/Loan%20Payment%20Schedule.ods HTTP/1.1" 200 244232
> 192.168.90.88 - - [18/Nov/2016:13:00:16 -0800] "POST /index.php/heartbeat HTTP/1.1" 200 4437
> 192.168.90.88 - craig [18/Nov/2016:13:00:21 -0800] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 4881
> 192.168.90.88 - craig [18/Nov/2016:13:00:32 -0800] "PROPFIND /remote.php/webdav/craig-Finances HTTP/1.1" 207 1237

3. Server logfile: ownCloud log (data/owncloud.log):

> {"reqId":"hbx+bWF8AUnfHYqrYaps","remoteAddr":"192.168.90.88","app":"files","message":" Backends pro
> vided no user object for ","level":3,"time":"2016-10-27T18:05:29+00:00"}
> {"reqId":"hbx+bWF8AUnfHYqrYaps","remoteAddr":"192.168.90.88","app":"core","message":"Error while ru
> nning background job (class: OC\\Command\\CommandJob, arguments: O:33:\"OCA\\Files_Versions\\Comman
> d\\Expire\":2:{s:43:\"\u0000OCA\\Files_Versions\\Command\\Expire\u0000fileName\";s:30:\"\/craig-Fin
> ances\/Budget2016.ods\";s:39:\"\u0000OCA\\Files_Versions\\Command\\Expire\u0000user\";s:5:\"craig\"
> ;}): {\"Exception\":\"OC\\\\User\\\\NoUserException\",\"Message\":\"Backends provided no user objec
> t for \",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/www\\\/public\\\/owncloud\\\/apps\\\/files_versions\
> \\/lib\\\/storage.php(79): OC\\\\Files\\\\Filesystem::initMountPoints(false)\\n#1 \\\/home\\\/www\\
> \/public\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/storage.php(662): OCA\\\\Files_Versions\\\
> \Storage::getUidAndFilename('\\\/craig-Finances...')\\n#2 \\\/home\\\/www\\\/public\\\/owncloud\\\/
> apps\\\/files_versions\\\/command\\\/expire.php(60): OCA\\\\Files_Versions\\\\Storage::expire('\\\/
> craig-Finances...')\\n#3 \\\/home\\\/www\\\/public\\\/owncloud\\\/lib\\\/private\\\/command\\\/comm
> andjob.php(34): OCA\\\\Files_Versions\\\\Command\\\\Expire->handle()\\n#4 \\\/home\\\/www\\\/public
> \\\/owncloud\\\/lib\\\/private\\\/backgroundjob\\\/job.php(52): OC\\\\Command\\\\CommandJob->run('O
> :33:\\\"OCA\\\\\\\\Files...')\\n#5 \\\/home\\\/www\\\/public\\\/owncloud\\\/lib\\\/private\\\/backg
> roundjob\\\/queuedjob.php(42): OC\\\\BackgroundJob\\\\Job->execute(Object(OC\\\\BackgroundJob\\\\Jo
> bList), Object(OC\\\\Log))\\n#6 \\\/home\\\/www\\\/public\\\/owncloud\\\/cron.php(171): OC\\\\Backg
> roundJob\\\\QueuedJob->execute(Object(OC\\\\BackgroundJob\\\\JobList), Object(OC\\\\Log))\\n#7 {mai
> n}\",\"File\":\"\\\/home\\\/www\\\/public\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php\
> ",\"Line\":386}","level":3,"time":"2016-10-27T18:05:29+00:00"}

...
> {"reqId":"ulgWyW7DTvm0ORaz5CYa","remoteAddr":"192.168.90.88","app":"core","message":"Error while running background job (class: OC\\Command\\CommandJob, arguments: O:33:\"OCA\\Files_Versions\\Command\\Expire\":2:{s:43:\"\u0000OCA\\Files_Versions\\Command\\Expire\u0000fileName\";s:30:\"\/craig-Finances\/Budget2016.ods\";s:39:\"\u0000OCA\\Files_Versions\\Command\\Expire\u0000user\";s:5:\"craig\";}): {\"Exception\":\"OC\\\\User\\\\NoUserException\",\"Message\":\"Backends provided no user object for \",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/www\\\/public\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/storage.php(79): OC\\\\Files\\\\Filesystem::initMountPoints(false)\\n#1 \\\/home\\\/www\\\/public\\\/owncloud\\\/apps\\\/files_versions\\\/lib\\\/storage.php(662): OCA\\\\Files_Versions\\\\Storage::getUidAndFilename('\\\/craig-Finances...')\\n#2 \\\/home\\\/www\\\/public\\\/owncloud\\\/apps\\\/files_versions\\\/command\\\/expire.php(60): OCA\\\\Files_Versions\\\\Storage::expire('\\\/craig-Finances...')\\n#3 \\\/home\\\/www\\\/public\\\/owncloud\\\/lib\\\/private\\\/command\\\/commandjob.php(34): OCA\\\\Files_Versions\\\\Command\\\\Expire->handle()\\n#4 \\\/home\\\/www\\\/public\\\/owncloud\\\/lib\\\/private\\\/backgroundjob\\\/job.php(52): OC\\\\Command\\\\CommandJob->run('O:33:\\\"OCA\\\\\\\\Files...')\\n#5 \\\/home\\\/www\\\/public\\\/owncloud\\\/lib\\\/private\\\/backgroundjob\\\/queuedjob.php(42): OC\\\\BackgroundJob\\\\Job->execute(Object(OC\\\\BackgroundJob\\\\JobList), Object(OC\\\\Log))\\n#6 \\\/home\\\/www\\\/public\\\/owncloud\\\/cron.php(171): OC\\\\BackgroundJob\\\\QueuedJob->execute(Object(OC\\\\BackgroundJob\\\\JobList), Object(OC\\\\Log))\\n#7 {main}\",\"File\":\"\\\/home\\\/www\\\/public\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php\",\"Line\":386}","level":3,"time":"2016-11-17T06:13:13+00:00"}

**Comment by craigarno** [comment]
I might add a suggestion, whatever is done to cure this behavior with Calc ODS files should also be applied to all types of **[Open Document](https://en.wikipedia.org/wiki/OpenDocument)** _ISO/IEC 26300-3:2015 Information technology -- Open Document Format for Office Applications_ file types; i.e. ODT, ODS, ODP, ODB, ODG, ODF, OTT, OTS, OTP, OTG, ODM, OTM, OTH.  Or more generally; O[DT][TSPBGFMH].  This is a potentially incomplete list as I'm not an expert on these file types.

**Comment by craigarno** [comment]
I'm trying to collect more information for this issue, and now I'm somewhat baffled by OwnCloud behavior.
My assumption is owncloud decides on an update direction based on timestamp.
I opened up permissions on the Linux side (side which needs updating) and start with files which look like this:
> -rwxrw-rw-  1 craig users    43756 Nov 18 13:57 Budget2016.ods
> 11/24/2016  10:51 AM            43,790 Budget2016.ods

Based on timestamp, I would expect the Windows version to be copied to the Linux file system.
That isn't what happens.

What happens is OwnCloud copies the older Linux file down to the Windows file system with the message under "_Not Synced_": "**Server version downloaded, copied changed local file into conflict file**"

When I look at the Windows file system, there exists the older Server version of the file and a "conflict" file with my most recent changes:
> 11/18/2016  01:57 PM            43,756 Budget2016.ods
> 11/24/2016  10:51 AM            43,790 Budget2016_conflict-20161118-135725.ods

I expected the newer Windows file to be copied to the Linux file system. So I'm baffled.

**Comment by guruz** [comment]
Do you remember in which client version the sync worked?

Is the directory on the server shared as read only or something like that?

Your server log contains something about the versions app, I wonder if it can be related to this? @PVince81 

The update direction is not based on timestamp but on the server state stored in the client's .csync_journal.db by comparing this with the real server state on sync. Based on this it can be decided if something changed compared to the last saved state in the DB.

FYI @ogoffart

**Comment by craigarno** [comment]
I just rechecked this issue with file Budget2016.ods and it now appears to work as expected.  The things which changed since my report are; OwnCloud is now version 8.2.9 (was 8.2.8), LibreOffice is now 5.3.03 x64 (was 5.1.5.2 x64).

Sorry for my late response, I wasn't notified of a new response on this issue. My settings suggest I should have been notified, but I wasn't.  So I'll answer your questions about my earlier post.

No I don't remember which Sync client version worked, but as I said my recollection is about 2-3 release versions prior to 2.2.4 build 6408 described above.

The OC setup
![oc finances link](https://cloud.githubusercontent.com/assets/4262930/23135138/3af75650-f74c-11e6-9150-8e4f11cc9118.jpg)
craig/Documents/Finances

The directory is a soft link to another directory, with the following permissions:

> Finances -> /home/usr/craig/quicken/
> drwxr-x---  28 craig users     4096 Feb 18 23:43 .
> -rw-r-----   1 craig users    43790 Nov 24 10:51 Budget2016.ods


I'm logged in as "craig", so I should have full R/W access to the file.  And as I explained earlier, it was working until about 1-2 client upgrades prior to my initial report.

With newer versions of software involved this issue does not reproduce.  Files Sync as I expect and as they did before.  File permissions and locations are the same as the original report, and Sync Client is still 2.2.4 build 6408.

Newer versions:
OwnCloud 8.2.9
LibreOffice 5.3.0.3 x64

I think it is safe to close this issue as non-repro.
